src/ensembl/production/metadata/api/genome.py

```diff
@@ -104,11 +141,12 @@ def fetch_genomes(self, genome_id=None, genome_uuid=None, assembly_accession=Non
                 .join(OrganismGroupMember.organism_group) \
                 .filter(OrganismGroup.type.in_(group_type)).filter(OrganismGroup.name.in_(group))
 
+        # Apply additional filters based on the provided parameters
```

Comment:
Have you tested these together? There are a few joins that might have to be done together before the filtering, but I am not sure.

Reply:
I added some tests and played with the code separately, I didn't get any unexpected behaviour